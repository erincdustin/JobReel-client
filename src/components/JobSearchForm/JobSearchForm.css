.JobSearchForm {
  width: 320px;
  border: 1px black solid;
  min-width: 300px;
  margin: 0 auto;
  padding: 5%;
  line-height: 150%;
  border-radius: 10px;
  text-align: center;
  margin: auto;
  background: #3b3a50;
  color: #eeebd9;
  box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
}

.jobSearchFormImage {
  position: relative;
  grid-column-start: 1;
  grid-column-end: 9;
}

.jobSearchFormImage > img {
  height: 320px;
  width: 320px;
  position: absolute;
  top: -100px;
  left: -15%;
  z-index: -1;
}