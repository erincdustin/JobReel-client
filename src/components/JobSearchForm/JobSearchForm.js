import React from 'react';
import { Input, Label } from '../../components/Form/Form'
import Button from '../../components/Button/Button'
import './JobSearchForm.css';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';

class JobSearchForm extends React.Component {
  static contextType = JobReelContext

  componentDidMount() {
    if (this.context.savedJobs.length === 0) {
      jobReelApiService.getSavedJobs()
        .then(res => {
          this.context.setSavedJobs(res);
        })
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    // const industry = e.target.industry.value;
    const jobTitle = e.target['job-title'].value;
    const location = e.target.location.value;
    this.context.setSearch({ jobTitle, location })
    this.props.history.push(`/jobslist`)
  }

  takeToNext = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/jobsearch/results';
    history.push(destination);
  }

  render() {
    return (
      <div className='job-search'>
        <div className='title'>
          <h2>Search for Job</h2>
        </div>
        <form className='JobSearchForm' onSubmit={this.handleSubmit}>
          {/* <div>
            <Label htmlFor='industry-input'>
              Industry
            </Label>
            <br/>
            <Input
              ref={this.firstInput}
              id='industry-input'
              name='industry'
            />
          </div> */}
          <div>
            <Label htmlFor='job-title-input'>
              Job Title
            </Label>
            <br />
            <Input
              id='job-title-input'
              name='job-title'
            />
          </div>
          <div>
            <Label htmlFor='location-input'>
              Location
            </Label>
            <br />
            <Input
              ref={this.firstInput}
              id='location-input'
              name='location'
            />
          </div>
          <br />
          <Button type="submit">Submit</Button>
        </form>
      </div>
    )
  }
}

export default JobSearchForm;