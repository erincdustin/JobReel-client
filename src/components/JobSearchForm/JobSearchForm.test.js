import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PropTypes from 'prop-types'
import {BrowserRouter} from 'react-router-dom'
import JobSearchForm from './JobSearchForm'

class ContextProvider extends React.Component{
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      savedJobs: [1, 2, 3]
    }
  })

  render(){
    return this.props.children
  }
}

describe('Job Search Form smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><JobSearchForm /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('Job Search Form snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><JobSearchForm /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})