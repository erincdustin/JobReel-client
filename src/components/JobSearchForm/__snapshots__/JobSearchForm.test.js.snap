// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Search Form snapshot test renders the UI as expected 1`] = `<JobSearchForm />`;
